propolis/src/common.rs

```diff
@@ -60,6 +72,20 @@ impl<'a> ReadOp<'a> {
             write_offset: 0,
         }
     }
+
+    /// Constructs a child read operation from within an existing read
+    /// operation.
+    ///
+    /// # Arguments
+    ///
+    /// - `op_offset`: Offset of the child operation. Does not need to correlate
+    /// to the `parent` operation's offset.
+    /// - `parent`: The operation from which this operation is being split.
+    /// - `range`: The location within the parent operation to be moved
+    /// to the child.
+    // TODO: Why doesn't this impact the `write_offset` of the parent, or take
+    // that into consideration? What if the parent already wrote past this
+    // region?
```

Comment:
I started tracking read_offset/write_offset in the hopes that I could, for example, verify that the data of the operation was entirely consumed when the emulation was finished.  That got a little tricky considering how the `new_child` behavior can split arbitrarily.  I didn't feel like tracking extents or a bitfield or something quite yet.  In this case, when something splits off a new child, it's expected that the consumer of that ReadOp/WriteOp will intend to consume the whole thing (and thus start from a zeroed read/write offset)